compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -30,6 +30,116 @@
 #include "il/Node_inlines.hpp"
 #include "infra/Bit.hpp"
 
+typedef enum RegisterStatus {
+   OK_FOR_REUSE,
+   DO_NOT_TOUCH
+} RegisterStatus;
+
+static inline RegisterStatus
+evaluateRegister(TR::Register **childReg, TR::Node *child, TR::Node *parent, TR::CodeGenerator *cg)
+   {
+   int64_t value;
+   RegisterStatus status;
+
+	if (child->getOpCode().isLoadConst() && child->getRegister() == NULL)
+   {
+      value = child->getLongInt();
+      if (constantIsUnsignedImm12(value))
+      {
+         //TODO: Cover special cases for immediate loads
+      }
+      else
+      {
+         *childReg = cg->allocateRegister();
+         loadConstant64(cg, parent, value, *childReg);
+         status = OK_FOR_REUSE;
+      }
+	}
+	else
+   {
+      *childReg = cg->evaluate(child);
+      status = DO_NOT_TOUCH;
+	}
+   return status;
+   }
+
+static inline TR::Register *
+evaluateBooleanTypeInteger(TR::Node *node, TR::InstOpCode::Mnemonic regOp, TR::CodeGenerator *cg, uint8_t sizeOfOperation)
+   {
+   TR::Register *srcReg[3];
+   RegisterStatus srcStatus[3];
+   TR::Node *children[3];
+   TR::Register *trgReg;
+   bool assigned = false;
+
+   if(sizeOfOperation > 0)
```

Comment:
Can `sizeOfOperation` ever be 0?  If so, what is the semantics of that?

Reply:
In our current version of this, if 0 is passed into the sizeOfOperation field when the second child in the operation is 0 (in idiv and ldiv), it will handle division by zero by simply allocating a register and returning it (it will skip creating any instructions at all because it never enters any for loops or ifs). We were wondering if you wanted that case to be handled elsewhere (in the evaluator functions themselves), or if it would be okay for the helper to do it for us.

Reply:
This isn't obvious from the implementation provided.  From the current implementation, it appears this parameter maps to the number of children on the node.  If that's the case, then passing in "0" is special case.  If you have newer code that shows what you have in mind then please push that.